runtime/sema/type.go

```diff
@@ -859,6 +872,18 @@ func (t *GenericType) Resolve(typeArguments *TypeParameterTypeOrderedMap) Type {
 	return ty
 }
 
+func (t *GenericType) Map(gauge common.MemoryGauge, f func(Type) Type) Type {
+	typeParameter := &TypeParameter{
+		Name:      t.TypeParameter.Name,
+		Optional:  t.TypeParameter.Optional,
+		TypeBound: t.TypeParameter.TypeBound.Map(gauge, f),
+	}
```

Comment:
I'm not quite sure if returning a new/different type parameter here is not going to work: function types refer to their type parameters and generic type occurrences in the type (e.g. parameter's type) refer to them by pointer.